### ISSUE TITLE
Need FAQ / support for Gutenberg / WordPress 5.0

### ISSUE THREAD

**Issue by bobbingwide** [futtta/autoptimize#214]
I'm using Gutenberg and developing my own blocks to replace shortcodes that I developed in my oik plugin.  I have a countdown block that uses jQuery to run the countdown timer. 

When Gutenberg is activated and Autoptimize's Optimize JavaScript code is checked, neither the shortcode nor the block operates as expected... the countdown is not displayed. 

There are many errors in the console log. 
![image](https://user-images.githubusercontent.com/2474435/48719171-ba115800-ec14-11e8-86ee-90c484a17934.png)

With WordPress 5.0-beta5, the result is similar.

![image](https://user-images.githubusercontent.com/2474435/48719455-7f5bef80-ec15-11e8-8271-fc3367766258.png)

Deselecting the checkbox resolves the issue.

But doing this means I lose a lot of the benefits of Autoptimize. 

Is this something that has been looked at? 
If so, is there a generic solution to the problem?
Is there one in the pipeline?

### Workaround
I can use the xsl:stylesheet trick to fool Autoptimize so that the site works regardless of the checkbox settings. But again, that's not really a solution to the problem.

**Comment by futtta** [comment]
I'm not 100% sure I understand the issue @bobbingwide ;

Some questions:
* Do you get the errors while working in Gutenberg (so as an admin) or while browsing the site (as a visitor, logged off)?
* I guess your shortcode/ block has JS dependencies, have you tried excluding them from JS optimization?

frank

**Comment by bobbingwide** [comment]
Hi Frank, I get them when viewing the contents while logged in or logged out.

Gutenberg / WordPress 5.0 has a lot of JavaScript. My blocks have JavaScript dependencies too.
When I deactivate my plugin then I don't get any problems. I'll try adding the dependency on lodash and see what happens.

**Comment by futtta** [comment]
> Gutenberg / WordPress 5.0 has a lot of JavaScript

it does, but on the admin-side, rarely on the frontend :-)

> I'll try adding the dependency on lodash and see what happens.

not sure what you'll need to exclude, but if for the sole purpose of showing a timer you're getting all those JS errors you are (or seem to be) introducing a lot of dependencies which I personally would prefer not to introduce in my frontend ;-)

**Comment by bobbingwide** [comment]
In my Gutenberg blocks plugin ( https://github.com/bobbingwide/oik-block ) I have built a number of blocks into one target file. The countdown block invokes a shortcode which enqueues some countdown.js. When Gutenberg is deactivated the shortcode is executed and the code works with Autoptimize. I’ll attempt to reduce the problem to the simplest.

**Comment by bobbingwide** [comment]
In my earlier tests I'd failed to find the Advanced settings. With WordPress 5.0-beta5, in order to use Autoptimize's JavaScript optimization I needed to select the first three check boxes
![image](https://user-images.githubusercontent.com/2474435/48772257-bf75ad80-ecbb-11e8-9b2c-07cf3eaa6fac.png)

When my plugin is activated a whole load of JavaScript is enqueued. 
![image](https://user-images.githubusercontent.com/2474435/48773237-b0dcc580-ecbe-11e8-8fcb-b7cb9db4be3d.png)

This happened regardless of the content being displayed.

I discovered that my plugins was enqueuing editor blocks in the front end. 
I've removed that logic and the problem has gone away.

Time will tell whether or not this error on my part is still worthy of a FAQ.
